zebra-network/src/peer/priority.rs

```diff
@@ -105,30 +105,47 @@ fn address_is_valid_for_outbound_connections(
         );
     }
 
-    // Ignore ports used by similar networks: Flux/ZelCash and misconfigured Zcash.
+    address_is_valid_for_inbound_listeners(peer_addr, network)
+}
+
+/// Is the supplied [`SocketAddr`] valid for inbound listeners on `network`?
+///
+/// This is used to check Zebra's configured Zcash listener port.
+pub fn address_is_valid_for_inbound_listeners(
+    listener_addr: &SocketAddr,
+    network: impl Into<Option<Network>>,
+) -> Result<(), &'static str> {
+    // TODO: make private IP addresses an error unless a debug config is set (#3117)
+
+    // Ignore ports used by potentially compatible nodes: misconfigured Zcash ports.
     if let Some(network) = network.into() {
-        if peer_addr.port() == network.default_port() {
+        if listener_addr.port() == network.default_port() {
             return Ok(());
         }
 
-        if peer_addr.port() == 8232 {
+        if listener_addr.port() == 8232 {
             return Err(
                 "invalid peer port: port is for Mainnet, but this node is configured for Testnet",
             );
-        } else if peer_addr.port() == 18232 {
+        } else if listener_addr.port() == 18232 {
             return Err(
                 "invalid peer port: port is for Testnet, but this node is configured for Mainnet",
             );
-        } else if peer_addr.port() == 18344 {
-            return Err(
-                "invalid peer port: port is for Regtest, but Zebra does not support that network",
-            );
-        } else if [16125, 26125].contains(&peer_addr.port()) {
-            // 16125/26125 is used by Flux/ZelCash, which uses the same network magic numbers as Zcash,
-            // so we have to reject it by port
-            return Err("invalid peer port: port is for a non-Zcash network");
         }
     }
 
+    // Ignore ports used by potentially compatible nodes: other coins and unsupported Zcash regtest.
+    if listener_addr.port() == 18344 {
+        return Err(
+            "invalid peer port: port is for Regtest, but Zebra does not support that network",
+        );
+    } else if [8033, 18033, 16125, 26125].contains(&listener_addr.port()) {
```

Comment:
Question: Our acceptance test use random ports. If one of these ports is chosen then the test will fail probably. Is this correct and accepted ?

Reply:
This is a good question!

In most tests, we ask the OS for an unallocated port in our tests, and it should choose from an ephemeral port range:
https://github.com/ZcashFoundation/zebra/blob/beb45fc514d78f91fde18a602d475ff55102c4fe/zebra-test/src/net.rs#L110-L132

The ephemeral port range is usually above 49,000.

If it's not, the probability of test failure would be approximately 6/65535 or 1 in 10,000 test functions. I don't think we've ever seen a failure like this happen, so it seems like the OS is choosing from the upper range.

Reply:
Ok, thats good to know, thanks.